client/src/components/JurisdictionAdmin/BatchRoundDataEntry.tsx

```diff
@@ -1,544 +1,659 @@
+    results: {},
+  }
 }
 
-const emptyTallySheet = (index: number) => ({
-  name: `Tally Sheet #${index}`,
-  results: {},
-})
+interface IProps {
+  electionId: string
+  jurisdictionId: string
+  roundId: string
+}
 
-const BatchTallySheetsModal = ({
+const BatchRoundDataEntry: React.FC<IProps> = ({
   electionId,
   jurisdictionId,
   roundId,
-  contest,
-  batch,
-  closeModal,
-}: {
-  electionId: string
-  jurisdictionId: string
-  roundId: string
-  contest: IContest
-  batch: IBatch
-  closeModal: () => void
 }) => {
+  const batchesQuery = useBatches(electionId, jurisdictionId, roundId)
+  const contestsQuery = useContestsJurisdictionAdmin(electionId, jurisdictionId)
   const recordBatchResults = useRecordBatchResults(
     electionId,
     jurisdictionId,
     roundId
   )
-  const initialTallySheets =
-    batch.resultTallySheets.length > 1
-      ? batch.resultTallySheets
-      : batch.resultTallySheets.concat([
-          emptyTallySheet(batch.resultTallySheets.length + 1),
-        ])
-  const { register, control, handleSubmit, formState, errors } = useForm<{
-    resultTallySheets: IBatchResultTallySheet[]
-  }>({
-    defaultValues: { resultTallySheets: initialTallySheets },
+  const { confirm, confirmProps } = useConfirm()
+
+  const [searchQuery, setSearchQuery] = useState('')
+  const [debouncedSearchQuery] = useDebounce(searchQuery)
+  const [selectedBatchId, setSelectedBatchId] = useState<IBatch['id']>()
+  const [selectedTabId, setSelectedTabId] = useState('')
+  const [isEditing, setIsEditing] = useState(false)
+  const [isRenamingSheet, setIsRenamingSheet] = useState(false)
+  const [sheetNameBeforeRename, setSheetNameBeforeRename] = useState<{
+    index: number
+    name: string
+  }>()
+
+  const batches = batchesQuery.isSuccess ? batchesQuery.data.batches : []
+  const filteredBatches = batches.filter(batch =>
+    batch.name.toLowerCase().includes(debouncedSearchQuery.toLowerCase())
+  )
+  const selectedBatch = batches.find(batch => batch.id === selectedBatchId)
+
+  const initialResultTallySheets = selectedBatch?.resultTallySheets.length
+    ? selectedBatch.resultTallySheets
+    : [constructEmptyResultTallySheet(0)]
+  const formMethods = useForm<IBatchRoundDataEntryFormState>({
+    defaultValues: {
+      resultTallySheets: initialResultTallySheets,
+    },
   })
-  const { fields, append, remove } = useFieldArray({
+  const {
+    control,
+    formState,
+    getValues,
+    handleSubmit,
+    reset,
+    setValue,
+    watch,
+  } = formMethods
+  const {
+    append: appendSheet,
+    fields: sheets,
+    remove: removeSheet,
+  } = useFieldArray<IBatchResultTallySheet>({
     control,
     name: 'resultTallySheets',
   })
 
-  const onSubmit = async ({
+  // Reset state whenever a new batch is selected
+  useEffect(() => {
+    reset({ resultTallySheets: initialResultTallySheets })
+    setSelectedTabId(VOTE_TOTALS_TAB_ID)
+    setIsEditing(false)
+  }, [selectedBatchId])
```

Comment:
I think you might be able to remove this effect if you made a subcomponent for a single batch (the entire contents of the detail view, including the whole form).